File: ems_hr_web/src/main/webapp/js/service/salaryService.js
//服务层
app.service('salaryService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../salary/findAll.do');		
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../salary/findPage.do?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../salary/findOne.do?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../salary/add.do',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../salary/update.do',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../salary/delete.do?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../salary/search.do?page='+page+"&rows="+rows, searchEntity);
	}    	
});

File: ems_interface/src/main/java/com/ems/service/SalaryService.java
package com.ems.service;
import java.util.List;
import com.ems.pojo.Salary;

import entity.PageResult;
/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface SalaryService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<Salary> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum, int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(Salary salary);
	
	
	/**
	 * 修改
	 */
	public void update(Salary salary);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public Salary findOne(Integer id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Integer[] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(Salary salary, int pageNum, int pageSize);
	
}
